code/controllers/subsystem/jobs.dm

```diff
@@ -574,36 +574,58 @@ SUBSYSTEM_DEF(jobs)
-	var/datum/money_account/M = create_account(H.real_name, rand(50,500)*10, null)
-	var/remembered_info = ""
+	var/datum/money_account/account
+	if(job?.department_account_access)
+		account = GLOB.station_money_database.create_account(H.real_name, COMMAND_MEMBER_STARTING_BALANCE, ACCOUNT_SECURITY_ID, "NAS Trurl Accounting", TRUE)
+	else
+		account = GLOB.station_money_database.create_account(H.real_name, CREW_MEMBER_STARTING_BALANCE, ACCOUNT_SECURITY_ID, "NAS Trurl Accounting", TRUE)
+
+	for(var/datum/job_objective/objective as anything in H.mind.job_objectives)
+		objective.owner_account = account
 
-	remembered_info += "<b>Your account number is:</b> #[M.account_number]<br>"
-	remembered_info += "<b>Your account pin is:</b> [M.remote_access_pin]<br>"
-	remembered_info += "<b>Your account funds are:</b> $[M.money]<br>"
+	var/remembered_info = ""
+	remembered_info += "<b>Your account number is:</b> #[account.account_number]<br>"
+	remembered_info += "<b>Your account pin is:</b> [account.account_pin]<br>"
 
-	if(M.transaction_log.len)
-		var/datum/transaction/T = M.transaction_log[1]
-		remembered_info += "<b>Your account was created:</b> [T.time], [T.date] at [T.source_terminal]<br>"
 	H.mind.store_memory(remembered_info)
+	H.mind.initial_account = account
+
+	//add them to their department datum, (this relates a lot to money account I promise)
+	var/list/users_departments = get_departments_from_job(job.title)
+	for(var/datum/station_department/department as anything in users_departments)
+		var/datum/department_member/member = new()
+		member.name = H.real_name
+		member.role = job.title
+		member.member_account = account
+		member.can_approve_crates = job?.department_account_access
+		department.members += member
+
+	to_chat(H, "<span class='boldnotice'>Your account number is: [account.account_number], your account pin is: [account.account_pin]</span>")
 
 	// If they're head, give them the account info for their department
-	if(job && job.head_position)
-		remembered_info = ""
-		var/datum/money_account/department_account = GLOB.department_accounts[job.department]
+	if(!job?.department_account_access)
+		return
+
+	remembered_info = ""
+	//we use the first entry in the list because it prioritizes generic departments over command
+	for(var/datum/station_department/department as anything in users_departments)
+		if(job.title != department.head_of_staff && job.title != "Quartermaster")
+			continue
+		var/datum/money_account/department_account = department.department_account
+		if(!department_account)
+			return
 
-		if(department_account)
-			remembered_info += "<b>Your department's account number is:</b> #[department_account.account_number]<br>"
-			remembered_info += "<b>Your department's account pin is:</b> [department_account.remote_access_pin]<br>"
-			remembered_info += "<b>Your department's account funds are:</b> $[department_account.money]<br>"
+		remembered_info += "<b>As a head of staff you have access to your department's funds<br>"
```

Comment:
You need to close this bold tag, it's causing all head of staff antags to have fully bolded memories and antag objectives.